### PR TITLE
Checkout: Change cached contact details to use useQuery instead of Redux

### PR DESCRIPTION
**Pull request by sirbrillig**
## Proposed Changes

This PR refactors the `useCachedContactDetails()` to use `useQuery` instead of the Redux state for cached contact details. This will help decouple checkout from Redux so it can more easily be used outside of calypso.

As part of this process, this PR also removes contact details from Redux state. This was relatively easy because there are only two places contact data is used: the domain management form, which uses a separate Redux Whois data store, and checkout, which uses a "managed" contact form where the data is kept in a separate `@wordpress/data` store. The only thing actually using the removed data was the hook which prefills the contact fields in checkout.

To clarify some of this removal, this PR refactors the ccTLD contact forms (.ca, .fr, and .uk) into TypeScript and removes all the parts that were unused.

## Testing Instructions

- Use an account which has cached contact details. Verify that, without this PR, when you visit checkout the billing info step gets auto-completed.
- On this branch, verify that when you visit checkout the billing info step gets auto-completed.

Because of the ccTLD refactors, it's also important to test that they still work.

- Add a .ca, .fr, or .uk domain to your shopping cart.
- Visit checkout and fill out the contact details step, including the additional ccTLD fields.
- Click to complete the contact details step.
- Verify there are no errors. **NOTE:** ccTLD form fields are not pre-populated in checkout before or after this diff.

### CODE REVIEW

File: client/components/domains/register-domain-step/index.jsx [hunk 1]
```diff
@@ -522,7 +521,6 @@ class RegisterDomainStep extends Component {
 					{ this.renderFilterContent() }
 					{ this.renderDomainExplanationImage() }
 					{ this.renderSideContent() }
-					<QueryContactDetailsCache />
```
Comment (sirbrillig) [inline client/components/domains/register-domain-step/index.jsx:525]: From what I can tell, this was never used. There are no selectors for the domain contact details in the register domain step.

File: client/components/data/domain-management/index.jsx [hunk 2]
```diff
@@ -37,13 +35,11 @@ class DomainManagementData extends Component {
...
 
 	render() {
-		const { needsContactDetails, needsDomains, needsPlans, needsProductsList, selectedSite } =
-			this.props;
+		const { needsDomains, needsPlans, needsProductsList, selectedSite } = this.props;
 
 		return (
 			<div>
 				<PageViewTracker path={ this.props.analyticsPath } title={ this.props.analyticsTitle } />
-				{ selectedSite && needsContactDetails && <QueryContactDetailsCache /> }
```
Comment (sirbrillig) [inline client/components/data/domain-management/index.jsx:46]: From what I can tell, this was never used. There are no selectors for the domain contact details in the domain-management pages; they use the Whois data instead.

File: client/components/domains/registrant-extra-info/ca-form.tsx [hunk 3]
```diff
@@ -104,34 +111,27 @@ export class RegistrantExtraInfoCaForm extends PureComponent {
...
-		const newContactDetails = {};
+		const newContactDetails: DomainContactDetails = {};
 
 		newContactDetails.extra = {
 			ca: { [ camelCase( id ) ]: type === 'checkbox' ? checked : value },
 		};
 
-		this.props.updateContactDetailsCache( { ...newContactDetails } );
-
-		if ( this.props.isManaged ) {
```
Comment (sirbrillig) [inline client/components/domains/registrant-extra-info/ca-form.tsx:121]: All uses of the ccTLD forms were using `isManaged` so it has been made the default.

File: client/my-sites/plans/main.jsx [hunk 4]
```diff
@@ -419,7 +418,6 @@ class Plans extends Component {
 				{ selectedSite.ID && <QuerySitePurchases siteId={ selectedSite.ID } /> }
 				<DocumentHead title={ translate( 'Plans', { textOnly: true } ) } />
 				<PageViewTracker path="/plans/:site" title="Plans" />
-				<QueryContactDetailsCache />
```
Comment (sirbrillig) [inline client/my-sites/plans/main.jsx:422]: From what I can tell, nothing uses contact details on the plans page.

**Comment by jjchrisdiehl** [comment]
Functional testing:

**Regular contact info test**
Works as described from Calypso.live link:
image

**.fr domain contact info test**
No errors on completion:
image

**Pull request merged**